neo/Core/InvocationTransaction.cs

```diff
@@ -58,10 +61,39 @@ public override JObject ToJson()
             return json;
         }
 
-        public override bool Verify(IEnumerable<Transaction> mempool)
+        public override bool Verify(IEnumerable<Transaction> mempool, InteropService service = null)
         {
             if (Gas.GetData() % 100000000 != 0) return false;
-            return base.Verify(mempool);
+            return base.Verify(mempool, service);
+        }
+
+        protected override bool VerifyReceivingScripts(InteropService service = null)
+        {
+            HashSet<UInt160> contracts = new HashSet<UInt160>();
+            foreach (UInt160 hash in Outputs.Select(p => p.ScriptHash).Distinct())
+            {
+                ContractState contract = Blockchain.Default.GetContract(hash);
+                if (contract == null) continue;
+                if (!contract.Payable) return false;
+                contracts.Add(hash);
+            }
+
+            if (contracts.Count > 0) {
+                HashSet<UInt160> calledContracts = new HashSet<UInt160>();
+                using (StateReader readerService = new StateReader())
+                {
+                    ApplicationEngine engine = new ApplicationEngine(TriggerType.VerificationR, this, Blockchain.Default, service == null ? readerService : service, Gas);
```

Comment:
If `Gas` is a very large number, and the `Script` is an infinite loop, then this attack will be zero cost.